File: src/services/siteService.ts
import api from './api';
import type { Site, PaginatedResponse, PaginationParams } from '@/types';

export const siteService = {
  async getAll(params?: Partial<PaginationParams>): Promise<PaginatedResponse<Site>> {
    const { data } = await api.get('/sites', { params });
    return data;
  },

  async getById(id: string): Promise<Site> {
    const { data } = await api.get(`/sites/${id}`);
    return data;
  },

  async getByCustomer(customerId: string): Promise<Site[]> {
    const { data } = await api.get(`/customers/${customerId}/sites`);
    return data;
  },

  async create(site: Partial<Site>): Promise<Site> {
    const { data } = await api.post('/sites', site);
    return data;
  },

  async update(id: string, site: Partial<Site>): Promise<Site> {
    const { data } = await api.put(`/sites/${id}`, site);
    return data;
  },

  async delete(id: string): Promise<void> {
    await api.delete(`/sites/${id}`);
  },
};

File: FieldServiceManangement-main/src/main/java/com/fieldservicemanagement/field_service_management/service/SiteService.java
package com.fieldservicemanagement.field_service_management.service;

import com.fieldservicemanagement.field_service_management.entity.Customer;
import com.fieldservicemanagement.field_service_management.entity.Site;
import com.fieldservicemanagement.field_service_management.repository.CustomerRepository;
import com.fieldservicemanagement.field_service_management.repository.SiteRepository;
import org.springframework.stereotype.Service;

import java.util.List;
import java.util.NoSuchElementException;

@Service
public class SiteService {

    private final SiteRepository siteRepository;
    private final CustomerRepository customerRepository;

    public SiteService(SiteRepository siteRepository, CustomerRepository customerRepository) {
        this.siteRepository = siteRepository;
        this.customerRepository = customerRepository;
    }

    public List<Site> getAll() {
        return siteRepository.findAll();
    }

    public Site getById(Long id) {
        return siteRepository.findById(id)
                .orElseThrow(() -> new NoSuchElementException("Site not found with id: " + id));
    }

    public List<Site> getByCustomerId(Long customerId) {
        return siteRepository.findByCustomerId(customerId);
    }

    public Site create(Site site) {
        return siteRepository.save(site);
    }

    public Site update(Long id, Site siteDetails) {
        Site site = getById(id);
        site.setName(siteDetails.getName());
        site.setAddress(siteDetails.getAddress());
        site.setCustomer(siteDetails.getCustomer());
        return siteRepository.save(site);
    }

    public void delete(Long id) {
        siteRepository.deleteById(id);
    }
}